lib/stream/xlsx/worksheet-reader.js

```diff
@@ -87,14 +89,35 @@ class WorksheetReader extends EventEmitter {
     _.each(this._keys, f);
   }
 
-  // =========================================================================
-  // Read
+  async read() {
+    try {
+      for await (const events of this.parse()) {
+        for (const {eventType, value} of events) {
+          this.emit(eventType, value);
+        }
+      }
+      this.emit('finished');
+    } catch (error) {
+      this.emit('error', error);
+    }
+  }
 
-  _emitRow(row) {
-    this.emit('row', row);
+  async *[Symbol.asyncIterator]() {
+    for await (const events of this.parse()) {
+      const rows = [];
+      for (const {eventType, value} of events) {
+        if (eventType === 'row') {
+          rows.push(value);
```

Comment:
hmm... will it has performance effort if here you put:
```javascript
    yield value;
```

It should make these simpler:
```javascript
for await (const worksheetReader of workbookReader) {
  for await (const row of worksheetReader) {
//without this:    for (const row of rows) {
      // ...
    //}
  }
}
```

as you mention, that should be slower, but if could I ask you to check it again and show me benchmark result?
I a little bit worry about backward compatibility it the future when we would keep this API.